src/DefaultEventLogger.ts

```diff
@@ -54,6 +55,37 @@ class DefaultEventLogger implements EventLogger, EventPreProcessor, EventPostPro
       return result
     }
 
+    createNewTraceMetadata(service: string, sampled?: number | undefined, flags?: number | undefined, timestamp?: string | undefined): EventTraceMetadata {
+
+      let newMeta = new EventTraceMetadata(service, new Uuid(), new Uuid(), undefined, sampled, flags, timestamp);
+      return newMeta;
+    }
+
+    createChildTraceMetadata(parentTraceMetadata: EventTraceMetadata): EventTraceMetadata {
+      return new EventTraceMetadata(parentTraceMetadata.service, parentTraceMetadata.traceId, new Uuid(), parentTraceMetadata.spanId)
+    }
+
+    logNewTraceForMessageEnvelope(messageEnvelope: any, service: string, sampled?: number | undefined, flags?: number | undefined, timestamp?: string | undefined): EventMessage {
```

Comment:
I am not sure if we need to separate operations for new vs child for this. 

It should just require a check on the tracing context. E.g. is there a traceId, and is there a spanId in the "injected" trace. If there isn't, then it can be assumed that its a new trace, otherwise it must be a child.

Let me know your thoughts here.

Reply:
We have two operations mainly because the "logChild" operation inherits the service property and doesn't set any other prop. It is also related to the previous comment.

Regarding the second comment, both `traceId` and `spanId` are mandatory. So we can't have a  case with a valid `traceId` and no `spanId`.  We also need to support the creation of a "sibling" trace, ( same `traceId`, different `spanId`, no `parentId` ). I didn't provide an operation for this case.

I can redesign the API to:
- include optional properties in the args list
- Have 3 operations:
- logNewTraceForMessageEnvelope 
- logChildTraceForMessageEnvelope
- logSiblingTraceForMessageEnvelope

Regarding the parameter "parentTraceMetadata", I think I can remove it and pick it up from the `messageEnvelope` as `messageEnvelope.metadata.event`


I would also add the more primitive operations:
- createNewTraceMetadata
- createChildTraceMetadata
- createSiblingTraceMetadata

comments?

Reply:
Implemented in version `6.4.6-snapshot`